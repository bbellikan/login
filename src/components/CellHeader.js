import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class CellHeader extends Component {
    constructor(props) {
        super(props)

        this.dragstart_handler = this.dragstart_handler.bind(this);
        this.drop_handler = this.drop_handler.bind(this);
        this.dragover_handler = this.dragover_handler.bind(this);
        this.mousedown_handler = this.mousedown_handler.bind(this);
        this.mouseclicksort_handler = this.mouseclicksort_handler.bind(this);
        this.mouseclickdelete_handler = this.mouseclickdelete_handler.bind(this);
    }

    mouseclicksort_handler(ev) {
        this.props.updateSort(this.props.gridCol.number);
    }

    mouseclickdelete_handler(ev) {
        this.props.deleteCol(this.props.gridCol.number);
    }

    dragstart_handler(ev) {
        let dragData = {
            colNumber: this.props.gridCol.number
        };

        ev.dataTransfer.setData("text/plain", JSON.stringify(dragData));
        ev.dataTransfer.dropEffect = "copy";
    }

    dragover_handler(ev) {
        ev.preventDefault();
        ev.dataTransfer.dropEffect = "move"
    }

    drop_handler(ev) {
        let dragData = JSON.parse(ev.dataTransfer.getData("text"));
        this.props.updateColOrder(dragData.colNumber, this.props.gridCol.order);
    }

    mousedown_handler(ev) {
        const rect = ev.target.getBoundingClientRect();

        let data = {
            targetid: ev.target.id,
            clientX: ev.clientX,
            width: rect.width,
            isResizing: this.props.onRightEdge(ev)
        };

        this.props.updateClicked(data);
    }

    render() {
        let sortDirection = this.props.gridCol.sort==='a' ?
            <div onClick={this.mouseclicksort_handler}>
                {' ' + this.props.gridCol.header + ' ' } &#708;
            </div>
            : this.props.gridCol.sort==='d' ?
                <div onClick={this.mouseclicksort_handler}>
                    {' ' + this.props.gridCol.header + ' ' } &#709;
                </div>
                : <div onClick={this.mouseclicksort_handler}>
                    {' ' + this.props.gridCol.header}
                </div>;


            return (

                <div id={this.props.gridCol.number + 'header'}
                     className="cell cell-header"
                     draggable="true"
                     onDragStart={this.dragstart_handler}
                     onDrop={this.drop_handler}
                     onDragOver={this.dragover_handler}
                     onMouseDown={this.mousedown_handler}

                >
                    <div onClick={this.mouseclickdelete_handler}>
                         <i className="fa fa-times"/>
                    </div> {sortDirection}

                </div>
            )
    }
}

CellHeader.propTypes = {
    gridCol:PropTypes.object.isRequired,
    updateClicked:PropTypes.func.isRequired,
    onRightEdge:PropTypes.func.isRequired,
    updateColOrder:PropTypes.func.isRequired,
    updateSort : PropTypes.func.isRequired,
    deleteCol : PropTypes.func.isRequired
};


