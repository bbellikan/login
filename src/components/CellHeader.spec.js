import React from 'react';
import { shallow } from 'enzyme';
import CellHeader from './CellHeader';

const setup = (setupProps = {}) => {
    const defaultProps = {
        gridCol:{
            "number": 0,
            "order": 0,
            "header": "type",
            "sort": "n",
            "deleted": false,
            "type": "STRING",
            "inputType": "text",
            "width": 150
        },
        updateClicked:jest.fn(),
        onRightEdge:jest.fn(),
        updateColOrder:jest.fn(),
        updateSort : jest.fn(),
        deleteCol : jest.fn(),
    };

    const props = {...defaultProps, ...setupProps};
    const wrapper = shallow(
        <CellHeader {...props}/>
    );

    return {
        props,
        wrapper
    };
};

describe('CellHeader', () => {
    it('renders without crashing', () => {
        const {wrapper} = setup();
        expect(wrapper).toMatchSnapshot();
    });

});
