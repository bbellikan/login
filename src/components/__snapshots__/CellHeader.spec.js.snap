// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellHeader renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CellHeader
    deleteCol={[Function]}
    gridCol={
        Object {
            "deleted": false,
            "header": "type",
            "inputType": "text",
            "number": 0,
            "order": 0,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          }
    }
    onRightEdge={[Function]}
    updateClicked={[Function]}
    updateColOrder={[Function]}
    updateSort={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          onClick={[Function]}
>
          <i
                    className="fa fa-times"
          />
</div>,
        " ",
        <div
          onClick={[Function]}
>
           type
</div>,
      ],
      "className": "cell cell-header",
      "draggable": "true",
      "id": "0header",
      "onDragOver": [Function],
      "onDragStart": [Function],
      "onDrop": [Function],
      "onMouseDown": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa fa-times"
/>,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-times",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " type",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " type",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            onClick={[Function]}
>
            <i
                        className="fa fa-times"
            />
</div>,
          " ",
          <div
            onClick={[Function]}
>
             type
</div>,
        ],
        "className": "cell cell-header",
        "draggable": "true",
        "id": "0header",
        "onDragOver": [Function],
        "onDragStart": [Function],
        "onDrop": [Function],
        "onMouseDown": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-times"
/>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-times",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " type",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " type",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
